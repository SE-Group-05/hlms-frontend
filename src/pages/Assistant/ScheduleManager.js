import React from "react";
import { Box, Center, Heading, Wrap, WrapItem,Divider } from "@chakra-ui/react";
import ScheduleCard from "../../components/Assistant/ScheduleCard";
import SearchBar from "../../components/Assistant/SearchBar";

function ScheduleManager() {
    return (
        <>
            <Center w="100%" h="auto" p={4} pb='100px'>
                <Box w="90%" h="auto" borderWidth="1px" borderRadius="lg" p='30px'>
                    <Heading ml='20px' mb='20px'>Schedule Manager</Heading>
                    <Center >
                    <SearchBar text='Search Schedules...' />
                    </Center>
                    <Divider/>
                    <Heading as="h2" size="lg" ml='20px' mb='20px' mt='30px'>Schedules</Heading>
                    <Box>
                        <Wrap spacing="30px">
                            <WrapItem>
                                <ScheduleCard place='Mirissa' user='Kamal perera' date='2021/01/20' travelmethod='bus' fair='Rs. 200.00' state='completed' path={['Mirissa', 'Galle fort', 'Jungle beach']} />
                            </WrapItem>
                             <WrapItem>
                                <ScheduleCard place='Mirissa' user='Kamal perera' date='2021/01/20' travelmethod='bus' fair='Rs. 200.00' state='completed' path={['Mirissa', 'Galle fort', 'Jungle beach']} />
                            </WrapItem>
                             <WrapItem>
                                <ScheduleCard place='Mirissa' user='Kamal perera' date='2021/01/20' travelmethod='bus' fair='Rs. 200.00' state='completed' path={['Mirissa', 'Galle fort', 'Jungle beach']} />
                            </WrapItem>
                             <WrapItem>
                                <ScheduleCard place='Mirissa' user='Kamal perera' date='2021/01/20' travelmethod='bus' fair='Rs. 200.00' state='completed' path={['Mirissa', 'Galle fort', 'Jungle beach']} />
                            </WrapItem>
                             <WrapItem>
                                <ScheduleCard place='Mirissa' user='Kamal perera' date='2021/01/20' travelmethod='bus' fair='Rs. 200.00' state='completed' path={['Mirissa', 'Galle fort', 'Jungle beach']} />
                            </WrapItem>
                             <WrapItem>
                                <ScheduleCard place='Mirissa' user='Kamal perera' date='2021/01/20' travelmethod='bus' fair='Rs. 200.00' state='completed' path={['Mirissa', 'Galle fort', 'Jungle beach']} />
                            </WrapItem>
                        </Wrap>
                    </Box>
                </Box>
                
            </Center>
                    
            
        </>
    )
    
}

export default ScheduleManager;