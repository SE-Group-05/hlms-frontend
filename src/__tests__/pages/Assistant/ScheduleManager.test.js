// import dependencies
import React from "react";

// import react-testing methods
import { render, fireEvent, cleanup } from "@testing-library/react";

// add custom jest matchers from jest-dom
import "@testing-library/jest-dom/extend-expect";

// import react dom
import ReactDOM from "react-dom";

// import renderer for take snapshots
import renderer from "react-test-renderer";

// import the component for testing
import ScheduleManager from "../../../pages/Assistant/ScheduleManager";
import { MemoryRouter } from "react-router-dom";

afterEach(cleanup);


it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <MemoryRouter><ScheduleManager /></MemoryRouter>,
    div
  );
});

it("search button works correctly", async () => {
    const { queryByTestId } = render(
        <MemoryRouter><ScheduleManager /></MemoryRouter>
    );
    fireEvent.change(queryByTestId("search"),{ target: { value: 'mirissa' } })
    
  });
  


it("matches snapshot", () => {
  const tree = renderer.create(
    <MemoryRouter><ScheduleManager /></MemoryRouter>
  );

  expect(tree).toMatchSnapshot();
});
