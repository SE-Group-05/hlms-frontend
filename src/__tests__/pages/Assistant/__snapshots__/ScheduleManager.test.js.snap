// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="css-1pyso0s"
>
  <div
    className="css-102niab"
  >
    <h2
      className="chakra-heading css-1h40200"
    >
      Schedule Manager
    </h2>
    <div
      className="css-gmuwbf"
    >
      <div
        className="css-1ws0jr4"
      >
        <div
          className="chakra-input__group css-4302v8"
        >
          <div
            className="chakra-input__right-element css-a6fysx"
          >
            <button
              aria-label="Search database"
              className="chakra-button css-1kgd4fd"
              type="button"
            >
              <svg
                aria-hidden={true}
                className="chakra-icon css-onkibi"
                focusable={false}
                viewBox="0 0 24 24"
              >
                <path
                  d="M23.384,21.619,16.855,15.09a9.284,9.284,0,1,0-1.768,1.768l6.529,6.529a1.266,1.266,0,0,0,1.768,0A1.251,1.251,0,0,0,23.384,21.619ZM2.75,9.5a6.75,6.75,0,1,1,6.75,6.75A6.758,6.758,0,0,1,2.75,9.5Z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </div>
          <input
            className="chakra-input css-0"
            data-testid="search"
            name="search"
            onBlur={[Function]}
            onFocus={[Function]}
            placeholder="Search Schedules..."
          />
        </div>
      </div>
    </div>
    <hr
      aria-orientation="horizontal"
      className="chakra-divider css-1upb9tn"
      role="separator"
    />
    <h2
      className="chakra-heading css-hkdy83"
    >
      Schedules
    </h2>
    <div
      className="css-0"
    >
      <div
        className="chakra-wrap css-0"
      >
        <ul
          className="chakra-wrap__list css-yp34du"
        >
          <li
            className="chakra-wrap__listitem css-zji64q"
          >
            <div
              className="css-136pjoj"
            >
              <div
                className="chakra-stack css-c0d8a5"
              >
                <span
                  className="chakra-badge css-p5nc7k"
                >
                  Mirissa
                </span>
                <span
                  className="chakra-badge css-166rvjf"
                >
                  completed
                </span>
              </div>
              <div
                className="css-61ehcq"
              >
                <div
                  className="css-0"
                >
                  <p
                    className="chakra-text css-1lj023j"
                  >
                     Schedule By,
                  </p>
                  <p
                    className="chakra-text css-1lj023j"
                    data-testid="user"
                  >
                    <span
                      className="chakra-avatar css-1ohlb5u"
                    >
                      <div
                        aria-label="user"
                        className="chakra-avatar__initials css-0"
                      >
                        u
                      </div>
                    </span>
                    Kamal perera
                  </p>
                </div>
                <div
                  className="css-0"
                >
                  <p
                    className="chakra-text css-1lj023j"
                  >
                    Fair
                  </p>
                  <p
                    className="chakra-text css-6dvxm2"
                    data-testid="fair"
                  >
                    Rs. 
                    Rs. 200.00
                  </p>
                </div>
              </div>
              <hr
                aria-orientation="horizontal"
                className="chakra-divider css-1upb9tn"
                role="separator"
              />
              <p
                className="chakra-text css-8j1nyb"
              >
                Path
              </p>
              <div
                className="chakra-stack css-19kpmhn"
              >
                <span
                  className="chakra-badge css-ifsvay"
                  data-testid="path"
                >
                  Mirissa
                </span>
                <span
                  className="chakra-badge css-ifsvay"
                  data-testid="path"
                >
                  Galle fort
                </span>
                <span
                  className="chakra-badge css-ifsvay"
                  data-testid="path"
                >
                  Jungle beach
                </span>
              </div>
              <div
                className="css-61ehcq"
              >
                <div
                  className="css-0"
                >
                  <p
                    className="chakra-text css-8j1nyb"
                  >
                    Travel Method
                  </p>
                  <span
                    className="chakra-badge css-ifsvay"
                    data-testid="method"
                  >
                    bus
                  </span>
                </div>
                <div
                  className="css-0"
                >
                  <p
                    className="chakra-text css-8j1nyb"
                  >
                    Date
                  </p>
                  <span
                    className="chakra-badge css-ifsvay"
                  >
                    2021/01/20
                  </span>
                </div>
              </div>
            </div>
          </li>
          <li
            className="chakra-wrap__listitem css-zji64q"
          >
            <div
              className="css-136pjoj"
            >
              <div
                className="chakra-stack css-c0d8a5"
              >
                <span
                  className="chakra-badge css-p5nc7k"
                >
                  Mirissa
                </span>
                <span
                  className="chakra-badge css-166rvjf"
                >
                  completed
                </span>
              </div>
              <div
                className="css-61ehcq"
              >
                <div
                  className="css-0"
                >
                  <p
                    className="chakra-text css-1lj023j"
                  >
                     Schedule By,
                  </p>
                  <p
                    className="chakra-text css-1lj023j"
                    data-testid="user"
                  >
                    <span
                      className="chakra-avatar css-1ohlb5u"
                    >
                      <div
                        aria-label="user"
                        className="chakra-avatar__initials css-0"
                      >
                        u
                      </div>
                    </span>
                    Kamal perera
                  </p>
                </div>
                <div
                  className="css-0"
                >
                  <p
                    className="chakra-text css-1lj023j"
                  >
                    Fair
                  </p>
                  <p
                    className="chakra-text css-6dvxm2"
                    data-testid="fair"
                  >
                    Rs. 
                    Rs. 200.00
                  </p>
                </div>
              </div>
              <hr
                aria-orientation="horizontal"
                className="chakra-divider css-1upb9tn"
                role="separator"
              />
              <p
                className="chakra-text css-8j1nyb"
              >
                Path
              </p>
              <div
                className="chakra-stack css-19kpmhn"
              >
                <span
                  className="chakra-badge css-ifsvay"
                  data-testid="path"
                >
                  Mirissa
                </span>
                <span
                  className="chakra-badge css-ifsvay"
                  data-testid="path"
                >
                  Galle fort
                </span>
                <span
                  className="chakra-badge css-ifsvay"
                  data-testid="path"
                >
                  Jungle beach
                </span>
              </div>
              <div
                className="css-61ehcq"
              >
                <div
                  className="css-0"
                >
                  <p
                    className="chakra-text css-8j1nyb"
                  >
                    Travel Method
                  </p>
                  <span
                    className="chakra-badge css-ifsvay"
                    data-testid="method"
                  >
                    bus
                  </span>
                </div>
                <div
                  className="css-0"
                >
                  <p
                    className="chakra-text css-8j1nyb"
                  >
                    Date
                  </p>
                  <span
                    className="chakra-badge css-ifsvay"
                  >
                    2021/01/20
                  </span>
                </div>
              </div>
            </div>
          </li>
          <li
            className="chakra-wrap__listitem css-zji64q"
          >
            <div
              className="css-136pjoj"
            >
              <div
                className="chakra-stack css-c0d8a5"
              >
                <span
                  className="chakra-badge css-p5nc7k"
                >
                  Mirissa
                </span>
                <span
                  className="chakra-badge css-166rvjf"
                >
                  completed
                </span>
              </div>
              <div
                className="css-61ehcq"
              >
                <div
                  className="css-0"
                >
                  <p
                    className="chakra-text css-1lj023j"
                  >
                     Schedule By,
                  </p>
                  <p
                    className="chakra-text css-1lj023j"
                    data-testid="user"
                  >
                    <span
                      className="chakra-avatar css-1ohlb5u"
                    >
                      <div
                        aria-label="user"
                        className="chakra-avatar__initials css-0"
                      >
                        u
                      </div>
                    </span>
                    Kamal perera
                  </p>
                </div>
                <div
                  className="css-0"
                >
                  <p
                    className="chakra-text css-1lj023j"
                  >
                    Fair
                  </p>
                  <p
                    className="chakra-text css-6dvxm2"
                    data-testid="fair"
                  >
                    Rs. 
                    Rs. 200.00
                  </p>
                </div>
              </div>
              <hr
                aria-orientation="horizontal"
                className="chakra-divider css-1upb9tn"
                role="separator"
              />
              <p
                className="chakra-text css-8j1nyb"
              >
                Path
              </p>
              <div
                className="chakra-stack css-19kpmhn"
              >
                <span
                  className="chakra-badge css-ifsvay"
                  data-testid="path"
                >
                  Mirissa
                </span>
                <span
                  className="chakra-badge css-ifsvay"
                  data-testid="path"
                >
                  Galle fort
                </span>
                <span
                  className="chakra-badge css-ifsvay"
                  data-testid="path"
                >
                  Jungle beach
                </span>
              </div>
              <div
                className="css-61ehcq"
              >
                <div
                  className="css-0"
                >
                  <p
                    className="chakra-text css-8j1nyb"
                  >
                    Travel Method
                  </p>
                  <span
                    className="chakra-badge css-ifsvay"
                    data-testid="method"
                  >
                    bus
                  </span>
                </div>
                <div
                  className="css-0"
                >
                  <p
                    className="chakra-text css-8j1nyb"
                  >
                    Date
                  </p>
                  <span
                    className="chakra-badge css-ifsvay"
                  >
                    2021/01/20
                  </span>
                </div>
              </div>
            </div>
          </li>
          <li
            className="chakra-wrap__listitem css-zji64q"
          >
            <div
              className="css-136pjoj"
            >
              <div
                className="chakra-stack css-c0d8a5"
              >
                <span
                  className="chakra-badge css-p5nc7k"
                >
                  Mirissa
                </span>
                <span
                  className="chakra-badge css-166rvjf"
                >
                  completed
                </span>
              </div>
              <div
                className="css-61ehcq"
              >
                <div
                  className="css-0"
                >
                  <p
                    className="chakra-text css-1lj023j"
                  >
                     Schedule By,
                  </p>
                  <p
                    className="chakra-text css-1lj023j"
                    data-testid="user"
                  >
                    <span
                      className="chakra-avatar css-1ohlb5u"
                    >
                      <div
                        aria-label="user"
                        className="chakra-avatar__initials css-0"
                      >
                        u
                      </div>
                    </span>
                    Kamal perera
                  </p>
                </div>
                <div
                  className="css-0"
                >
                  <p
                    className="chakra-text css-1lj023j"
                  >
                    Fair
                  </p>
                  <p
                    className="chakra-text css-6dvxm2"
                    data-testid="fair"
                  >
                    Rs. 
                    Rs. 200.00
                  </p>
                </div>
              </div>
              <hr
                aria-orientation="horizontal"
                className="chakra-divider css-1upb9tn"
                role="separator"
              />
              <p
                className="chakra-text css-8j1nyb"
              >
                Path
              </p>
              <div
                className="chakra-stack css-19kpmhn"
              >
                <span
                  className="chakra-badge css-ifsvay"
                  data-testid="path"
                >
                  Mirissa
                </span>
                <span
                  className="chakra-badge css-ifsvay"
                  data-testid="path"
                >
                  Galle fort
                </span>
                <span
                  className="chakra-badge css-ifsvay"
                  data-testid="path"
                >
                  Jungle beach
                </span>
              </div>
              <div
                className="css-61ehcq"
              >
                <div
                  className="css-0"
                >
                  <p
                    className="chakra-text css-8j1nyb"
                  >
                    Travel Method
                  </p>
                  <span
                    className="chakra-badge css-ifsvay"
                    data-testid="method"
                  >
                    bus
                  </span>
                </div>
                <div
                  className="css-0"
                >
                  <p
                    className="chakra-text css-8j1nyb"
                  >
                    Date
                  </p>
                  <span
                    className="chakra-badge css-ifsvay"
                  >
                    2021/01/20
                  </span>
                </div>
              </div>
            </div>
          </li>
          <li
            className="chakra-wrap__listitem css-zji64q"
          >
            <div
              className="css-136pjoj"
            >
              <div
                className="chakra-stack css-c0d8a5"
              >
                <span
                  className="chakra-badge css-p5nc7k"
                >
                  Mirissa
                </span>
                <span
                  className="chakra-badge css-166rvjf"
                >
                  completed
                </span>
              </div>
              <div
                className="css-61ehcq"
              >
                <div
                  className="css-0"
                >
                  <p
                    className="chakra-text css-1lj023j"
                  >
                     Schedule By,
                  </p>
                  <p
                    className="chakra-text css-1lj023j"
                    data-testid="user"
                  >
                    <span
                      className="chakra-avatar css-1ohlb5u"
                    >
                      <div
                        aria-label="user"
                        className="chakra-avatar__initials css-0"
                      >
                        u
                      </div>
                    </span>
                    Kamal perera
                  </p>
                </div>
                <div
                  className="css-0"
                >
                  <p
                    className="chakra-text css-1lj023j"
                  >
                    Fair
                  </p>
                  <p
                    className="chakra-text css-6dvxm2"
                    data-testid="fair"
                  >
                    Rs. 
                    Rs. 200.00
                  </p>
                </div>
              </div>
              <hr
                aria-orientation="horizontal"
                className="chakra-divider css-1upb9tn"
                role="separator"
              />
              <p
                className="chakra-text css-8j1nyb"
              >
                Path
              </p>
              <div
                className="chakra-stack css-19kpmhn"
              >
                <span
                  className="chakra-badge css-ifsvay"
                  data-testid="path"
                >
                  Mirissa
                </span>
                <span
                  className="chakra-badge css-ifsvay"
                  data-testid="path"
                >
                  Galle fort
                </span>
                <span
                  className="chakra-badge css-ifsvay"
                  data-testid="path"
                >
                  Jungle beach
                </span>
              </div>
              <div
                className="css-61ehcq"
              >
                <div
                  className="css-0"
                >
                  <p
                    className="chakra-text css-8j1nyb"
                  >
                    Travel Method
                  </p>
                  <span
                    className="chakra-badge css-ifsvay"
                    data-testid="method"
                  >
                    bus
                  </span>
                </div>
                <div
                  className="css-0"
                >
                  <p
                    className="chakra-text css-8j1nyb"
                  >
                    Date
                  </p>
                  <span
                    className="chakra-badge css-ifsvay"
                  >
                    2021/01/20
                  </span>
                </div>
              </div>
            </div>
          </li>
          <li
            className="chakra-wrap__listitem css-zji64q"
          >
            <div
              className="css-136pjoj"
            >
              <div
                className="chakra-stack css-c0d8a5"
              >
                <span
                  className="chakra-badge css-p5nc7k"
                >
                  Mirissa
                </span>
                <span
                  className="chakra-badge css-166rvjf"
                >
                  completed
                </span>
              </div>
              <div
                className="css-61ehcq"
              >
                <div
                  className="css-0"
                >
                  <p
                    className="chakra-text css-1lj023j"
                  >
                     Schedule By,
                  </p>
                  <p
                    className="chakra-text css-1lj023j"
                    data-testid="user"
                  >
                    <span
                      className="chakra-avatar css-1ohlb5u"
                    >
                      <div
                        aria-label="user"
                        className="chakra-avatar__initials css-0"
                      >
                        u
                      </div>
                    </span>
                    Kamal perera
                  </p>
                </div>
                <div
                  className="css-0"
                >
                  <p
                    className="chakra-text css-1lj023j"
                  >
                    Fair
                  </p>
                  <p
                    className="chakra-text css-6dvxm2"
                    data-testid="fair"
                  >
                    Rs. 
                    Rs. 200.00
                  </p>
                </div>
              </div>
              <hr
                aria-orientation="horizontal"
                className="chakra-divider css-1upb9tn"
                role="separator"
              />
              <p
                className="chakra-text css-8j1nyb"
              >
                Path
              </p>
              <div
                className="chakra-stack css-19kpmhn"
              >
                <span
                  className="chakra-badge css-ifsvay"
                  data-testid="path"
                >
                  Mirissa
                </span>
                <span
                  className="chakra-badge css-ifsvay"
                  data-testid="path"
                >
                  Galle fort
                </span>
                <span
                  className="chakra-badge css-ifsvay"
                  data-testid="path"
                >
                  Jungle beach
                </span>
              </div>
              <div
                className="css-61ehcq"
              >
                <div
                  className="css-0"
                >
                  <p
                    className="chakra-text css-8j1nyb"
                  >
                    Travel Method
                  </p>
                  <span
                    className="chakra-badge css-ifsvay"
                    data-testid="method"
                  >
                    bus
                  </span>
                </div>
                <div
                  className="css-0"
                >
                  <p
                    className="chakra-text css-8j1nyb"
                  >
                    Date
                  </p>
                  <span
                    className="chakra-badge css-ifsvay"
                  >
                    2021/01/20
                  </span>
                </div>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
